### ISSUE TITLE
How to set Passenger max-pool-size

### ISSUE THREAD

**Issue by gene-miller** [bitnami/bitnami-docker-discourse#164]
After performing some load testing on Bitnami Discourse with 400+ concurrent users I found it was critical to be able to set the Passenger "max-pool-size" command line variable.

The only place I found this could be set was by editing the `/root/.nami/components/com.bitnami.discourse/nami.json` file and adding `--max-pool-size 12` to the `/bundle exec passenger start` command.
After running `nami restart discourse` the desired changes take affect.

This is obviously not ideal as the changes will not persist through a container restart.

Is there another way to control this setting? If not could this be exposed via an environment variable in a future release?

Thanks for all your hard work.

**Comment by dani8art** [comment]
Hi @gene-miller, thank you for the suggestion, it could be interesting to support this feature, it is cool. We have opened an internal task in order to evaluate it, unfortunately, due to we are a small team and regarding priorities and internal milestones we can't give you an ETA. However, if it is important for you and you feel you can handle it and you have a chance to create a PR we would be happy to handle it. 

I can give you some tips for managing this:

Discourse configuration is handle by a `nami` module. You can find the configuration scripts and the `nami`'s files here:

```console
ls -la .nami/components/com.bitnami.discourse/
```

The main components of a `nami` module are:
- `.nami/components/com.bitnami.discourse/main.js` where the logic is implemented, this is executed at container runtime and will configure the Discourse instance.
- `.nami/components/com.bitnami.discourse/nami.json` where some metadata and properties are stored. 
- `/discourse-inputs.json` this file maps the environment variables to `nami` module properties.

So if you want to add a new property you would need to do the following:

1. Add a new property to `nami.json`, e.g. `passengerMaxPoolSize`
2. Add logic in `main.js` as if `$app.passengerMaxPoolSize` has value then append it to the start command. There are some utils modules in `nami` like `$file`, `$os`, etc... 
3. Add a mapping between env vars and `nami` properties in `discourse-inputs.json`

I hope you are finally encouraged to do it.

**Comment by gene-miller** [comment]
Thanks @dani8art I am certainly keen to to submit a pull request, however these files don't seem to exist in this bitnami-docker-discourse repository. Perhaps they are included in with the dockerfile line

```
RUN bitnami-pkg unpack discourse-2.5.1-1 --checksum e4dedbce8d4ce723cbb3507aa8c5e95535193de0eb9c7d248a9a62df34d5e151
```

If this is the case perhaps there is a different repo that handles the bitnami discourse package?
Any assistance greatly appreciated

**Comment by dani8art** [comment]
Yes, you are right, we handle bitnami discourse package internally but you can modify it using docker volumes and paste the diff for these files here in a comment.

```yaml
  discourse:
    image: 'docker.io/bitnami/discourse:2-debian-10'
    ports:
      - '80:3000'
    depends_on:
      - postgresql
      - redis
    volumes:
      - 'discourse_data:/bitnami'
      - /path/to/development/nami.json:/root/.nami/components/com.bitnami.discourse/nami.json
    environment:
      - POSTGRESQL_HOST=postgresql
      - POSTGRESQL_ROOT_USER=postgres
      - POSTGRESQL_CLIENT_CREATE_DATABASE_NAME=bitnami_application
      - POSTGRESQL_CLIENT_CREATE_DATABASE_USERNAME=bn_discourse
      - POSTGRESQL_CLIENT_CREATE_DATABASE_PASSWORD=bitnami1
      - DISCOURSE_POSTGRESQL_NAME=bitnami_application
      - DISCOURSE_POSTGRESQL_USERNAME=bn_discourse
      - DISCOURSE_POSTGRESQL_PASSWORD=bitnami1
      - DISCOURSE_HOSTNAME=www.example.com
```

```diff
--- a/nami.json
+++ b/nami.json
@@ -137,6 +137,10 @@
       "description": "Directory to backup application folders",
       "default": "/bitnami/discourse"
     },
+    "passengerMaxPoolSize": {
+      "description": "Max Pool Size to be set on start command",
+      "default": ""
+    },
     "monitFile": {
       "value": "extra/monit.conf"
     },
```

**Comment by marcosbc** [comment]
Hi, FYI we just released a new container version, [2.7.0-debian-10-r4](https://github.com/bitnami/bitnami-docker-discourse#270-debian-10-r4). The initialization logic has received a major refactor, and its scripts are now located in the [`rootfs/` directory](https://github.com/bitnami/bitnami-docker-discourse/blob/master/2/debian-10/rootfs/).

In this new image version, we have implemented `DISCOURSE_PASSENGER_EXTRA_FLAGS` which will allow you to specify the Passenger max-pool-size option to start it.
